src/com/spikes2212/genericsubsystems/commands/MoveLimitedSubsystemWithPID.java

```diff
@@ -1,291 +1,48 @@
-
-    // Called just before this Command runs the first time
-    protected void initialize() {
-        movmentControl = new PIDController(KP, KI, KD, source, limitedSubsystem::tryMove);
-        movmentControl.setAbsoluteTolerance(tolerance);
-        movmentControl.setSetpoint(this.setpoint.get());
-        movmentControl.setOutputRange(-1, 1);
-        movmentControl.enable();
-    }
-
-    // Called repeatedly when this Command is scheduled to run
-    protected void execute() {
-        double newSetPoint = setpoint.get();
-        if (movmentControl.getSetpoint() != newSetPoint)
-            movmentControl.setSetpoint(newSetPoint);
-    }
-
-    // Make this return true when this Command no longer needs to run execute()
-    protected boolean isFinished() {
-        if (!movmentControl.onTarget()) {
-            lastTimeNotOnTarget = Timer.getFPGATimestamp();
-        }
-        return Timer.getFPGATimestamp() - lastTimeNotOnTarget >= waitTime;
-    }
-
-    // Called once after isFinished returns true
-    protected void end() {
-        movmentControl.disable();
-        limitedSubsystem.stop();
-    }
-
-    // Called when another command which requires one or more of the same
-    // subsystems is scheduled to run
-    protected void interrupted() {
-        end();
-    }
-
-}
+package com.spikes2212.genericsubsystems.commands;
+
+import java.util.function.Supplier;
+
+import com.spikes2212.genericsubsystems.LimitedSubsystem;
+
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.PIDSource;
+import com.spikes2212.utils.PIDSettings;
+
+/**
+ * This command moves a {@link LimitedSubsystem} using wpilib's
+ * {@link PIDController}. It also waits a specified amount of time after the
+ * error is within the given tolerance before stopping the PID Loop to make sure
+ * the subsystem doesn't go past and remain beyond the setpoint.
+ *
+ * @author Omri "Riki" Cohen and Itamar
+ * @see LimitedSubsystem
+ * @see PIDController
+ * @see PIDSettings
+ */
+
+public class MoveLimitedSubsystemWithPID extends MoveBasicSubsystemWithPID {
+
+	private LimitedSubsystem limitedSubsystem;
+
+	public MoveLimitedSubsystemWithPID(LimitedSubsystem limitedSubsystem, PIDSource source, Supplier<Double> setpoint,
+			PIDSettings PIDSettings) {
+		super(limitedSubsystem, source, setpoint, PIDSettings);
+		this.limitedSubsystem = limitedSubsystem;
+	}
+
+	public MoveLimitedSubsystemWithPID(LimitedSubsystem limitedSubsystem, PIDSource source, double setpoint,
+			PIDSettings PIDSettings) {
+		super(limitedSubsystem, source, setpoint, PIDSettings);
+	}
+
+	@Override
+	protected void initialize() {
+		movmentControl = new PIDController(PIDSettings.getKP(), PIDSettings.getKI(), PIDSettings.getKD(), source,
+				limitedSubsystem::tryMove);
+		movmentControl.setAbsoluteTolerance(PIDSettings.getTolerance());
+		movmentControl.setSetpoint(this.setpoint.get());
+		movmentControl.setOutputRange(-1, 1);
+		movmentControl.enable();
+	}
+

```

Comment:
where is the rest of the command? you can't use the isFinished from moveBasicSubsystemWithPID, because you need to consider the canMove.
and you made variable limitedSubsystem but you already have the BasicSubsystem variable in the moveBasicSubsystemWithPID command